src/main/java/com/fasterxml/jackson/core/util/ReadConstrainedTextBuffer.java

```diff
@@ -26,4 +26,13 @@ protected void validateStringLength(int length) throws StreamConstraintsExceptio
     {
         _streamReadConstraints.validateStringLength(length);
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void validateFPLength(int length) throws StreamConstraintsException
```

Comment:
Won't be needed, as per my comments (parsers will call `validateFPLength()` around same time as setting `TextBuffer` contents; and this only if number decoded is lexically valid (will not reset to empty String).